### ISSUE TITLE
BC25 Deployment fails with dependent apps in database

### ISSUE THREAD

**Issue by PeterConijn** [HodorNV/ALOps#804]
**Describe the bug**
Since we moved to BC25 on our SaaS acceptance environments, our deployments fail when there are dependent apps in the database. It keeps retrying a few times, but keeps failing. If I manually uninstall and unpublish all the apps, the deployments succeed without a hitch. 

I have run some tests with just two of our apps: our own "Dysel System App", which is the lowest tier, and our own "Dysel W1 Base App" which has a dependency on that system app. Neither has any other dependency except on BC itself.

Starting situation: 
    BC25.0 SaaS Sandbox (US)
    Dysel System App (v25.0) installed
    Dysel W1 (v25.0) installed
    Apps to be deployed are both of a higher revision (i.e. higher version no.) than the currently installed apps.

- Uninstall/Unpublish all apps: deployment succeeds
- Uninstall/Unpublish all apps except our lowest tier app: deployment succeeds
- Uninstall/Unpublish all apps except our lowest tier app and one dependent app and only deploy lowest tier app: deployment fails
- Uninstall/Unpublish all apps except our lowest tier app and one dependent app and only deploy only dependent app: deployment succeeds

Just in case the issue was in our system app, I have also tried this in a three-app configuration with our NA localization (which is dependent on our W1 "base" app). The NA localization deploys without a hitch, but the W1 deployment will fail with similar errors.

Just offhand, I'd say that this might be related to Microsoft's new app restore strategy, but I cannot say that for certain. I am just wondering if we are the only ones encountering this issue. Hopefully, someone here has encountered and managed to solve it. 

The error when dependent apps are present is:
```Text
error AL1024: A package with publisher 'Dysel', name 'DYSEL System Application', and a version compatible with '25.0.0.0' could not be loaded. Symbols for the requested app DYSEL System Application by Dysel 25.0.0.0 could not be found among this tenant's installed apps. AppGroup Id 19176
```

The error always references the app that we are *trying* to install (both the newly installed app and the previous version are of version 25.0.202411; only the revision number is different). 

We have tried sync and forcesync, but to no avail.

**the used yaml**
```yaml
#Your build pipeline references an undefined variable named ‘azure_tenant_id’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
#Your build pipeline references an undefined variable named ‘Deployment-TenantId’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
Your build pipeline references an undefined variable named ‘Deployment-ClientID’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
#Your build pipeline references an undefined variable named ‘Deployment-ClientSecret’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
#Your build pipeline references an undefined variable named ‘Deployment-UserId’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
#Your build pipeline references an undefined variable named ‘Deployment-UserPassword’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972

steps:
- task: Hodor.hodor-alops.ALOpsExtensionAPI.ALOpsExtensionAPI@1
  displayName: 'Deploy to NA'

  inputs:
    interaction: publish
    api_endpoint: 'https://api.businesscentral.dynamics.com/v2.0/$(azure_tenant_id)/DyselNA-DEV/api'
    dev_endpoint: 'https://api.businesscentral.dynamics.com/v2.0/DyselNA-DEV/dev'
    dev_schemeupdatemode: forcesync
    azure_tenant_id: '$(Deployment-TenantId)'
    azure_app_client_id: '$(Deployment-ClientID)'
    azure_app_client_secret: '$(Deployment-ClientSecret)'
    username: '$(Deployment-UserId)'
    password: '$(Deployment-UserPassword)'
    artifact_filter: '*_APP.app'
    checksecondsdelay: 15
    maxtries: 180
    replacepackageid: true
  continueOnError: true
  condition: succeededOrFailed()
```

**the output**
Also the complete output is necessary for us to see what is going on. 
Too long to paste: here is the log file
[9_Deploy to NA.log](https://github.com/user-attachments/files/17505041/9_Deploy.to.NA.log)

**Expected behavior**
I would expect the deployment to succeed even if there are dependent apps, as long as those dependent apps are not otherwise incompatible.

**Screenshots**
N/A

**Additional context**
We have not updated any release pipeline and this consistently happens across all of them and on all environments (I have tested on three environments - two US country versions, and one NL country version). 

I have also posted this issue on Yammer in case anyone else is getting it: https://www.yammer.com/dynamicsnavdev/#/threads/show?threadId=3033157648547840

Edit: changed the Yammer link since I moved it to a new group.

**Comment by PeterConijn** [comment]
Apparently, more people are experiencing this. 

Apparently, by specifying the parameter "dependencyPublishingOption" with value "ignore" (instead of the default value: "default") it works again, but we don't have that option in the ALOps publish steps, I think.

**Comment by CarloAxians** [comment]
We have the same issue, only solutions so far, is uninstall and unpublish the dependent apps.

**Comment by waldo1001** [comment]
I'm assuming we're only talking "dev port deploy", correct?
We'll add this option!

**Comment by waldo1001** [comment]
Released in v1.467.6362

Parameter `dev_dependencypublishingoption` with options:

- "none" 
- "default"
- "ignore" 
- "strict"

**Comment by PeterConijn** [comment]
I was on vacation, but yes, that is correct. Thank you, Waldo!
